qiskit/pulse/model/pulse_target.py

```diff
@@ -20,15 +20,80 @@
 from qiskit.pulse.exceptions import PulseError
 
 
-class LogicalElement(ABC):
+class PulseTarget(ABC):
+    """Base class of pulse target.
+
+    When playing a pulse on a quantum hardware, one typically has to define on what hardware component
+    the pulse will be played, and the frame (frequency and phase) of the carrier wave.
+    :class:`PulseTarget` addresses the first of two, and identifies the component which is the target
+    of the pulse. Every played pulse and most other instructions are associated with a
+    :class:`PulseTarget` on which they are performed.
+    """
+
+    def __init__(self, hash_identifier):
```

Comment:
This enforces its subclass to define a single hashable object. Seems like this removes flexibility of class design. I think it's better to turn `__hash__` into abstract method, and let each subclass explicitly define hash within its scope. Is the precache critical for performance?

Reply:
Honestly, I just didn't want to have the class empty. I don't think the precache is crucial here.